report-api/src/main/java/no/sikt/nva/data/report/api/fetch/client/NviInstitutionReportClient.java

```diff
@@ -27,12 +39,42 @@ public NviInstitutionReportClient(AuthorizedBackendClient authorizedBackendClien
     }
 
     public String fetchReport(String reportingYear, String topLevelOrganization, String acceptHeader)
-        throws IOException, InterruptedException {
+        throws ApiGatewayException {
         var request = generateRequest(reportingYear, topLevelOrganization, acceptHeader);
+        return attempt(() -> executeRequest(request)).orElseThrow(
+            failure -> logAndCreateBadGatewayException(request.build().uri(), failure.getException()));
+    }
+
+    private ApiGatewayException logAndCreateBadGatewayException(URI uri, Exception e) {
+        LOGGER.error("Unable to reach upstream: {}", uri, e);
+        if (e instanceof InterruptedException) {
+            Thread.currentThread().interrupt();
+        } else if (e instanceof ApiGatewayException) {
+            return (ApiGatewayException) e;
+        }
+        return new BadGatewayException("Unable to reach upstream!");
+    }
+
+    private String executeRequest(Builder request)
+        throws IOException, InterruptedException, ApiGatewayException {
         var response = authorizedBackendClient.send(request, BodyHandlers.ofString(UTF_8));
+        if (!(HTTP_OK == (response.statusCode()))) {
```

Comment:
parens

Reply:
`Operator '!' cannot be applied to 'int'`